File: data/code_pairs/fortran/2412.f90
! Assuming the module definitions and other necessary implementations are correctly defined elsewhere.
module gsw_mod_toolbox
    use, intrinsic :: iso_fortran_env, only: real64
    implicit none
contains
    real(real64) function gsw_saar(p, long, lat)
        real(real64), intent(in) :: p, long, lat
        ! Dummy implementation for testing
        gsw_saar = 0.01_real64
    end function gsw_saar
end module gsw_mod_toolbox

module gsw_mod_error_functions
    use, intrinsic :: iso_fortran_env, only: real64
    implicit none
contains
    real(real64) function gsw_error_code(code, func_name)
        integer, intent(in) :: code
        character(len=*), intent(in) :: func_name
        ! Dummy implementation for testing
        print *, "Error in function", trim(func_name), ": code", code
        gsw_error_code = -9999.0_real64
    end function gsw_error_code
end module gsw_mod_error_functions

module gsw_mod_kinds
    use, intrinsic :: iso_fortran_env, only: real64 => real64
end module gsw_mod_kinds

program test_gsw_fdelta
    use, intrinsic :: iso_fortran_env, only: real64
    use gsw_mod_toolbox
    use gsw_mod_error_functions
    use gsw_mod_kinds
    implicit none

    real(real64) :: result

    ! Test values
    result = gsw_fdelta(1000.0_real64, 30.0_real64, -60.0_real64)
    print *, "gsw_fdelta(1000, 30, -60) =", result

contains

    function gsw_fdelta(p, long, lat) result(fdelta)
        real(real64), intent(in) :: p, long, lat
        real(real64) :: fdelta
        real(real64) :: saar
        real(real64), parameter :: gsw_error_limit = 0.02_real64
        character(len=*), parameter :: func_name = "gsw_fdelta"

        saar = gsw_saar(p, long, lat)

        if (saar > gsw_error_limit) then
            fdelta = gsw_error_code(1, func_name)
        else
            fdelta = ((1.0_real64 + 0.35_real64) * saar) / (1.0_real64 - 0.35_real64 * saar)
        end if
    end function gsw_fdelta

end program test_gsw_fdelta

File: data/code_pairs/cpp/2412.cpp
#include <iostream>
#include <cmath>
#include <string>

// Assuming these functions are defined elsewhere correctly.
double gsw_saar(double p, double lon, double lat) {
    // Dummy implementation for testing
    return 0.01;
}

double gsw_error_code(int code, const std::string &func_name) {
    // Dummy implementation for testing
    std::cerr << "Error in function " << func_name << ": code " << code << std::endl;
    return std::nan("");
}

double gsw_fdelta(double p, double lon, double lat) {
    const std::string func_name = "gsw_fdelta";
    const double gsw_error_limit = 0.02;
    double saar = gsw_saar(p, lon, lat);
    
    if (saar > gsw_error_limit) {
        return gsw_error_code(1, func_name);
    } else {
        return ((1.0 + 0.35) * saar) / (1.0 - 0.35 * saar);
    }
}

int main() {
    double result = gsw_fdelta(1000.0, 30.0, -60.0);
    std::cout << "gsw_fdelta(1000, 30, -60) = " << result << std::endl;
    return 0;
}